### PR TITLE
Feature/refactor processors srp

### PR DESCRIPTION
**Pull request by jeffsuperglide**
S3 downloading files moved to geo-processor `handler` module removing that responsibility from each processor.  Processors can now be used in unit testing.  Python's `unittest` has been added to test processors.  These tests are implemented in a separate branch (not pushed yet) with a single product file for each processor.  Considering a separate `test data` repository and maybe include with HEC's [test data](https://github.com/HydrologicEngineeringCenter/dss-test-data/tree/master) repo.

### CODE REVIEW

**Comment by adamscarberry** [comment]
@jeffsuperglide I think you are missing the new ncep-stage4 processors (01h, 06h, 24h) in your branch.

**Comment by adamscarberry** [comment]
@jeffsuperglide  SNODAS is not working.  Please test with current data and historic.

**Comment by jeffsuperglide** [comment]
> @jeffsuperglide I think you are missing the new ncep-stage4 processors (01h, 06h, 24h) in your branch.

Seeing them [__here__](https://github.com/USACE/cumulus-api/tree/feature/refactor-processors-srp/async_geoprocess/src/cumulus_geoproc/processors)

File: async_geoprocess/src/cumulus_geoproc/utils/cgdal.py [hunk 1]
```diff
@@ -213,4 +251,4 @@ class GridProcess:
 
 
 if __name__ == "__main__":
-    pass
+    ...
```
Comment (brettpalmberg) [inline async_geoprocess/src/cumulus_geoproc/utils/cgdal.py]: I had to look this one up

File: async_geoprocess/src/cumulus_geoproc/utils/__init__.py [hunk 2]
```diff
@@ -31,31 +54,15 @@ def file_extension(file: str, preffix: str = "", suffix=".tif", maxsplit=-1):
...
-        ".tar",
-        ".bin",
-        ".grb",
-        ".zip",
-        ".bil",
-        ".dat",
-        ".txt",
-        ".tif",
-        ".tiff",
-        ".grb2",
```
Comment (brettpalmberg) [inline async_geoprocess/src/cumulus_geoproc/utils/__init__.py:48]: Please ensure .grb2 is included in the `EXTS = ` section. This was recently added for one of the new acquirables.

File: async_geoprocess/src/cumulus_geoproc/utils/__init__.py [hunk 3]
```diff
@@ -7,13 +7,34 @@
 import zipfile
 from cumulus_geoproc import logger
 
-# from cumulus_geoproc import logger
+EXTS = (
```
Comment (brettpalmberg) [inline async_geoprocess/src/cumulus_geoproc/utils/__init__.py:11]: Please ensure `.grb2` exists in this tuple. This was recently added for one of the new acquirables.

**Comment by brettpalmberg** [comment]
> S3 downloading files moved to geo-processor `handler` module removing that responsibility from each processor. Processors can now be used in unit testing. Python's `unittest` has been added to test processors. These tests are implemented in a separate branch (not pushed yet) with a single product file for each processor. Considering a separate `test data` repository and maybe include with HEC's [test data](https://github.com/HydrologicEngineeringCenter/dss-test-data/tree/master) repo.

Thanks @jeffsuperglide, looking at this now. Appreciate you tackling this, especially as we've been adding new processors in parallel.

**Comment by adamscarberry** [comment]
processor for stage4 06h not working, 01h and 24h work

File: async_geoprocess/src/cumulus_geoproc/processors/hrrr-total-precip.py [hunk 4]
```diff
@@ -27,27 +27,32 @@
 
 
 @pyplugs.register
-def process(src: str, dst: str, acquirable: str = None):
-    """Grid processor
+def process(*, src: str, dst: str = None, acquirable: str = None):
```
Comment (brettpalmberg) [inline async_geoprocess/src/cumulus_geoproc/processors/hrrr-total-precip.py:30]: Why do we pass `*` as an argument to all processor plugins? Is this needed?

**Review by brettpalmberg — CHANGES_REQUESTED** [review]

File: async_geoprocess/src/cumulus_geoproc/utils/__init__.py (hunk 3 above)
Comment (brettpalmberg) [reply async_geoprocess/src/cumulus_geoproc/utils/__init__.py:11]: I see .grb2 added 👍🏻 .

File: async_geoprocess/src/cumulus_geoproc/utils/__init__.py (hunk 2 above)
Comment (brettpalmberg) [reply async_geoprocess/src/cumulus_geoproc/utils/__init__.py:48]: @brettpalmberg made same comment in two places. Closing this one, discussion accounted for elsewhere.

File: async_geoprocess/src/cumulus_geoproc/processors/hrrr-total-precip.py (hunk 4 above)
Comment (jeffsuperglide) [reply async_geoprocess/src/cumulus_geoproc/processors/hrrr-total-precip.py:30]: Any argument after `*` forces a keyword only.  Just a way to make sure that the argument is assigned correctly.

**Comment by brettpalmberg** [comment]
closes USACE/cumulus#260

**Review by brettpalmberg — APPROVED** [review]

**Comment by brettpalmberg** [comment]
Marking my review complete, approving changes. cc @adamscarberry @jeffsuperglide

**Review by adamscarberry — APPROVED** [review]

**Pull request merged**